README.md

```diff
@@ -247,8 +257,82 @@ nginx config, deproxy response, addr and port can use templates
 in format `${part_variable}` where `part` is one of 'server',
 'tempesta', 'client' or 'backend'
 
+Warning: deproxy backend now is running on the Framework host, so
+framework ip should be specified in tempesta config.
+
 Example tests can be found in `selftests/test_framework.py`
 
 Tests can be skipped or marked as expected to fail.
 More info at [Python documentation](https://docs.python.org/3/library/unittest.html).
 
+## Internal structure and motivation of user configured tests
+
+User configured tests have very flexible structure. They allow arbitrary
+clients and server start, stop, making requests. This leads to several
+points in internal structure.
+
+### Using separate thread for polling cycle
+
+Now, deproxy client and deproxy server, all of them use the single polling
+cycle, as it was in functional tests. But we have differencies.
+
+We have 3 cases of using deproxy clients and server:
+
+1) both deproxy client and server are used
+
+2) only deproxy client is used
+
+3) only deproxy server is used
+
+First case corresponds to functional tests. The second and third have no
+corresponding case in old testing framework.
```

Comment:
> The second and third have no corresponding case in old testing framework.

May be it's because there is no need to use for them? I can imagine only two groups of tests:
1. Functional testing: It's important how very specific request (or response) is processed. In these tests we need to construct message, send it to Tempesta, and analyse how it's forwarded to target: which modifications was done to the message, which connection was used to forward the modified request and so on. It's about algorithm correctness.
2. Workload testing: we need to analyse correctness under load. No races happen, load distribution works as expected, correct request-response sequence, and many questions. It's not required to deeply analyse messages coming from Tempesta in this type of tests. Functional testing are much better to suit this goal.

Note, in old tests workload tests was called as `Stress tests`. It's not an accurate naming and it lead to many misunderstandings. Workload means enough load to test multythreaded code, but not in extreme conditions.

Hence I don't understand why do we need to start `wrk` against `Deproxy.Server` or `Deproxy.Client` against `Nginx`. I don't see any case for it. Yes, the new style of the test description allows to define every physically possible combination. But if there is no reason to implement all that freedom. Only  necessary and sufficient features should be implemented. Please, don't make tests too generic, it's only tests for Tempesta. They must be straight forward, simple, and 100% predictable.

Reply:
For example, we have wrk client, and 2 servers. And we want to check requests, which goes to them.